.gitignore

```diff
@@ -1,3 +1,28 @@
+
+# OSX leaves these everywhere on SMB shares
+._*
+
+# OSX trash
+.DS_Store
+
+# Eclipse files
+.classpath
+.project
+.settings/**
+
+# Vim swap files
+*.swp
+
+# Files generated by JetBrains IDEs, e.g. IntelliJ IDEA
+.idea/
+*.iml
+out/
+
+# Vscode files
+.vscode/**
+
 target
+tmp
+/bin
```

Comment:
why remove `bin` but add `/bin` here?

Reply:
Copied from TiKV. We only need to ignore the `bin` directory in the project root.